### ISSUE TITLE
Lists/Arrays data type 

### ISSUE THREAD

**Issue by msbeck** [chrismaltby/gb-studio#1324]
**Is your feature request related to a problem? Please describe.**
It seems next to impossible to handle large amounts of trivial items in the GUI editor. For example, a grid of plants in a farming sim, or an array of pieces in a puzzle game, or the status of any other grid based game element. Actor limits make it such that you cannot have each plant or game piece and actor, and you don't need all the functionality of actors anyway.

**Describe the solution you'd like**
I would like the ability to define an array of items, such as booleans or numbers or coordinate pairs in the GUI using the events. Additionally, an easy way to loop through each item in said list.

Variables that I can reference such as Variable1[0] and Grid[x][y]. Then being able to initiate or otherwise interact with them like setting each one to false.

**Describe alternatives you've considered**
Thing like this cannot all be actors and cannot be handled as individual variables because it's challenging/impossible to iterate through them all. 

**Additional context**
Could be used for: a grid plot of plants in a farming sim (list of plants each with their state). An abstract strategy game board like checkers (storing the board with the state of which piece is where. A map of valid piece placement locations). Or generally any other application of lists which are a fundamental way of storing and traversing data.

**Comment by cybergrunge-ops** [comment]
i agree. although i disagree with the tag "enhancement" for this,  as its barely even an enhancement and more of a core functionality that is just missing. lets do it
